pkg/kwok/server/metrics.go

```diff
@@ -17,11 +17,48 @@ limitations under the License.
 package server
 
 import (
-	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"sigs.k8s.io/kwok/pkg/apis/internalversion"
+	"sigs.k8s.io/kwok/pkg/kwok/metrics"
 )
 
 // InstallMetrics registers the metrics handler on the given mux.
 func (s *Server) InstallMetrics() {
-	promHandler := promhttp.Handler()
-	s.restfulCont.Handle("/metrics", promHandler)
+	defaultMetrics := metrics.DefaultMetrics()
+	s.restfulCont.Handle("/metrics", defaultMetrics.Handler())
+	for _, m := range s.config.Metrics {
+		path, err := executePathTemplate(m.Spec.Path, m)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to execute path template: %v", err))
+		}
+		reg := metrics.CustomMetric(m.Name)
```

Comment:
The purpose of this is just to create an empty registry and install handler. The registry metric installation is lazy. It will install metrics as soon as the `node` comes up in order to evaluate labels correctly :relaxed: